lib/better_html/node_iterator/location.rb

```diff
@@ -4,6 +4,10 @@ class Location
       attr_accessor :start, :stop
 
       def initialize(document, start, stop, line = nil, column = nil)
+        raise ArgumentError, "start location #{start} is out of range for document of size #{document.size}" if start > document.size
+        raise ArgumentError, "stop location #{stop} is out of range for document of size #{document.size}" if stop > document.size
+        raise ArgumentError, "end of range must be greater than start of range (#{stop} < #{start})" if stop < start
```

Comment:
or equal to?

Reply:
the case where `start == stop` is valid, with 1 character is selected

Reply:
Yeah, but right now the error message suggests that it's not valid.

Reply:
oops, you are right